examples/fabric-sync/main.cpp

```diff
@@ -35,6 +40,21 @@ int main(int argc, char * argv[])
 
     VerifyOrDie(ChipLinuxAppInit(argc, argv) == 0);
 
+#if defined(ENABLE_CHIP_SHELL)
+#if CHIP_DEVICE_CONFIG_APP_PLATFORM_ENABLED
```

Comment:
Not sure I follow why there is a check for `CHIP_DEVICE_CONFIG_APP_PLATFORM_ENABLED`

Reply:
The Shell commission commands depend on examples/platform/linux/CommissionerMain.cpp to implement their functionalities, which require this flag. Without it, the following compilation errors will occur:

```
../../../examples/fabric-sync/third_party/connectedhomeip/examples/platform/linux/CommissionerMain.cpp: In constructor ‘PairingCommand::PairingCommand()’:
../../../examples/fabric-sync/third_party/connectedhomeip/examples/platform/linux/CommissionerMain.cpp:243:9: error: class ‘PairingCommand’ does not have any field named ‘mOnDeviceConnectedCallback’
  243 |         mOnDeviceConnectedCallback(OnDeviceConnectedFn, this),
```

Reply:
Would some sort of static assert here be better? Is the app going to have some functionality without being able to call `Shell::RegisterCommands();`?

Essentially if I enable `ENABLE_CHIP_SHELL`, but don't enable `CHIP_DEVICE_CONFIG_APP_PLATFORM_ENABLED`, why do we let things still compile that prevents `ENABLE_CHIP_SHELL` from working?